static/js/cms_project_detail.js

```diff
@@ -235,38 +235,125 @@ export function set_page_info_cms_project_detail(uuid) {
     qrcode.style = "width:100px; height:100px; margin-top:15px;";
     qrcode.makeCode(
       location.protocol +
-        "//" +
-        window.location.host +
-        "/tasks/" +
-        obj_task.uuid
+      "//" +
+      window.location.host +
+      "/tasks/" +
+      obj_task.uuid
     );
 
     var obj_div_des = document.createElement("div");
     obj_div_des.className = "col-md-4 mt-4 mt-md-0";
     var obj_p_name = document.createElement("p");
-    obj_p_name.innerHTML = "活動設計名稱: ";
+    obj_p_name.classList.add("mb-3");
+    obj_p_name.innerHTML = `<span style='font-size: 18px;'>活動設計名稱 (${obj_task.uuid}): </span><br/>`;
     var obj_span_name = document.createElement("span");
-    obj_span_name.innerHTML = obj_task.name;
+    obj_span_name.innerHTML = "<b style='font-size: 14px'>" + obj_task.name + "</b>";
+
     var obj_p_period = document.createElement("p");
+    obj_p_period.classList.add("mb-3");
     obj_p_period.innerHTML = "日期: ";
     var obj_span_period = document.createElement("span");
     obj_span_period.innerHTML = obj_task.period;
-    var obj_p_idea = document.createElement("p");
-    obj_p_idea.className = "small";
 
+    const obj_desc = document.createElement("p");
+    obj_desc.style = "max-height: 80px; overflow: hidden; text-overflow: ellipsis; display: -webkit-box; -webkit-line-clamp: 3; -webkit-box-orient: vertical;";
+    obj_desc.innerHTML = obj_task.overview;
+
+    const obj_btn_readmore = document.createElement("a");
+    obj_btn_readmore.href = "javascript:void(0);";
+    obj_btn_readmore.className = "";
+    obj_btn_readmore.textContent = "Read more...";
+
+    if (obj_desc.textContent.length > 44) {
+      obj_btn_readmore.style.display = "block";
+    } else {
+      obj_btn_readmore.style.display = "none";
+    }
+
+    const obj_modal = document.createElement("div");
```

Comment:
```suggestion
    const obj_modal =  createModalDialog();
```
把下面產生這些 element 的方法包裝到 createModalDialog 的 function

Reply:
@ShouShouRen 盡可能將產生 element 的流程，包裝在 function 裡面，如果生成的過程中 element 會需要一些值，就用參數傳入進去。

Reply:
已修改完成！！